stock_picking_propagate_scheduled_date/models/stock_move.py

```diff
@@ -24,10 +24,14 @@ def _propagate_date(self, new_date):
                     continue
                 if move_dest.id in already_propagated_ids:
                     continue
-                move_dest.date -= delta
+                move_dest.with_context(skip_date_propagation=True).write(
+                    {
+                        "date": move_dest.date - delta,
+                    }
+                )
 
     def write(self, vals):
         # propagate date changes in the stock move chain
-        if "date" in vals:
+        if "date" in vals and not self.env.context.get("skip_date_propagation", False):
```

Comment:
This fix is not correct. We want the date change to propagate beyond the 1st chained move.